### ISSUE TITLE
Chapter 11: Validation loss is very high

### ISSUE THREAD

**Issue by akjain90** [ageron/handson-ml#194]
HI,
I have a doubt in chapter 11 Jupiter notebook, in block 29 (just before 'Batch normalization' heading), I am trying to print the value of validation loss, but the value is coming out to be very high. I am pretty sure I am missing some theoretical concept, but I am not able to figure it out.

`0 Batch accuracy: 0.9 Validation accuracy: 0.9208 Validation loss 58497.7`
`5 Batch accuracy: 0.94 Validation accuracy: 0.956 Validation loss 61143.7`
`10 Batch accuracy: 1.0 Validation accuracy: 0.9666 Validation loss 60269.5`

I am doing the following:
```python
X_val_scaled = (mnist.validation.images - means) / stds
acc_val,loss_val = sess.run([accuracy,loss],feed_dict={X: X_val_scaled, y: mnist.validation.labels})
```

Thanks.

**Comment by ageron** [comment]
Hi @akjain90 ,

Thanks for your question. The loss used in this network is the mean cross-entropy. This is the most common cost function used when training a classification neural network. The mean cross entropy computes the negative logarithm of the estimated probability for the correct class (averaging over all instances), so if the estimated probability is very close to zero, then the loss can be arbitrarily high. In other words, the cross entropy strongly penalizes a network that is both very confident and wrong. So it's possible for the loss to be very high even if the accuracy is okay: it might mean that it's not wrong very often, but when it is, it's very confident, so it has a strong loss.

That said, it does seem way too high in this case, I don't see how it's possible to get such a good accuracy with such a high loss: the network would have to predict a probability so close to zero for the correct class in the cases where it's wrong that with the floating point rounding errors, it would basically be equal to 0.0, and you would get a NaN.

So I'm guessing there's something fishy. :)

Could you please check that the loss is defined as a mean:

```python
    loss = tf.reduce_mean(xentropy, name="loss")
```

and not as a sum:

```python
    loss = tf.reduce_sum(xentropy, name="loss")
```

And please check that you are printing `loss_val` and that it's not being overridden, that kind of thing.

Hope this helps,
Aurélien

**Comment by ageron** [comment]
Closing this issue, please reopen if the problem persists.
